apps/web/screens/UniversitySearch/UniversitySearch.tsx

```diff
@@ -0,0 +1,1252 @@
+  >([])
+  const [query, setQuery] = useState('')
+  const searchTermHasBeenInitialized = useRef(false)
+  const [filteredResults, setFilteredResults] = useState<
+    Array<Fuse.FuseResult<Program>>
+  >(
+    data.map((item: Program, index: number) => {
+      return { item, refIndex: index, score: 1 }
+    }),
+  )
+  const { linkResolver } = useLinkResolver()
+
+  const [filters, setFilters] = useState<FilterProps>(initialFilters)
+
+  const [gridView, setGridView] = useState<boolean>(true)
+
+  const [totalPages, setTotalPages] = useState<number>(
+    Math.ceil(data.length / ITEMS_PER_PAGE),
+  )
+
+  useEffect(() => {
+    setTotalPages(Math.ceil(filteredResults.length / ITEMS_PER_PAGE))
+  }, [filteredResults])
+
+  useEffect(() => {
+    const comp = localStorage.getItem('comparison')
+    const viewChoice = localStorage.getItem('viewChoice')
+
+    if (!!comp) {
+      const comparison = JSON.parse(comp)
+      setSelectedComparison(comparison)
+    }
+
+    if (!!viewChoice) {
+      setGridView(viewChoice === 'true' ? true : false)
+    }
+  }, [])
+
+  //Fuse options:
+  // isCaseSensitive: false,
+  // includeScore: false,
+  // shouldSort: true,
+  // includeMatches: false,
+  // findAllMatches: false,
+  // minMatchCharLength: 1,
+  // location: 0,
+  // distance: 100,
+  // useExtendedSearch: false,
+  // ignoreLocation: false,
+  // ignoreFieldNorm: false,
+  // fieldNormWeight: 1,
+
+  const fuseOptions = {
+    threshold: 0.2,
+    keys: [
+      'nameIs',
+      'departmentNameIs',
+      'descriptionIs',
+      'degreeType',
+      'modeOfDelivery',
+      'startingSemesterSeason',
+      'durationInYears',
+      'universityId',
+    ],
+  }
+
+  const fuseInstance = new Fuse(data, fuseOptions)
+
+  useEffect(() => {
+    let activeFiltersFound: Array<{ key: string; value: Array<string> }> = []
+    Object.keys(filters).forEach(function (key, index) {
+      let str = key as keyof typeof filters
+      if (filters[str].length > 0) {
+        activeFiltersFound.push({ key, value: filters[str] })
+      }
+    })
+    //if no filters are active, then show no products
+    if (query === '' && activeFiltersFound.length === 0) {
+      resetFilteredList()
+    } else {
+      const results = SearchProducts({
+        fuseInstance,
+        query,
+        activeFilters: activeFiltersFound,
+      })
+
+      setFilteredResults(results)
+    }
+  }, [filters, query])
+
+  const resetFilteredList = () => {
+    const resultProducts: Array<Fuse.FuseResult<Program>> = data.map(
+      (item: Program, index: number) => {
+        return { item, refIndex: index, score: 1 }
+      },
+    )
+    setFilteredResults(resultProducts)
+  }
+
+  const createPrimaryCTA = (item: Program) => {
+    const now = new Date()
+    const CTA: CTAProps = {
+      label: n('apply', 'Sækja um'),
+      variant: 'primary',
+      size: 'small',
+      icon: 'arrowForward',
+      iconType: 'outline',
+      onClick: () => console.log('TODO'),
```

Comment:
Implementation missing?

Reply:
There is no link to the application at the moment, that is in the next phase, that's why this is a TODO